### PR TITLE
cephadm: use importlib.metadata for querying ceph_iscsi's version 

### PR DESCRIPTION
**Pull request by tchaikov**
use importlib.metadata for querying ceph_iscsi's version and fallback to
pkg_resources. as the former is only available in Python 3.8, while
the latter is deprecated.

Refs https://tracker.ceph.com/issues/66201

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/cephadm/cephadmlib/daemons/iscsi.py [hunk 1]
```diff
@@ -119,22 +119,23 @@ def customize_container_binds(
...
-                ctx.container_engine.path,
-                'exec',
-                container_id,
-                '/usr/bin/python3',
-                '-c',
-                "import pkg_resources; print(pkg_resources.require('ceph_iscsi')[0].version)",
-            ],
-            verbosity=CallVerbosity.QUIET,
-        )
+        def python(s):
```
Comment (phlogistonjohn) [inline src/cephadm/cephadmlib/daemons/iscsi.py]: Good idea, but I think that it should be a file level function, as there's very little need for it to be a closure. Something like:
```
def _python(engine: str, container_id: str, src: str) -> Tuple[str, str, int]:
    ...
```

Comment (tchaikov) [reply src/cephadm/cephadmlib/daemons/iscsi.py]: the whole point of creating this helper is to make the caller's life easier, and to reduce the number of parameters, the reason why i made it a nested function is that it is not supposed to be shared across other functions. and i wanted to make it very clear. but yes, it does not have to be a closure. 

i am moving it to a `staticmethod`, hope this helps.

**Comment by tchaikov** [comment]
v2:

- rebased.

**Comment by tchaikov** [comment]
jenkins test api

**Review by phlogistonjohn — APPROVED** [review]
looks ok to me

**Review by adk3798 — APPROVED** [review]

File: src/cephadm/cephadmlib/daemons/iscsi.py (hunk 1 above)
Comment (tchaikov) [reply src/cephadm/cephadmlib/daemons/iscsi.py]: thank you John!

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-06-10_15:49:42-orch:cephadm-wip-adk-testing-2024-06-10-0759-distro-default-smithi/

failures consisted of

- 18 `in cluster log` type failures, still being cleaned up and a known issue
- 3 failures in the cleanup of rados_api_tests, don't know the tracker number but this is a known issue
- 2 failures in mds_upgrade_sequence, known issue
- 1 failure with staggered upgrade while agent is active, known issue
- lots of dead jobs due to `Error reimaging machines: Failed to power on smithiXXX`, general lab/teuthology issue unrelated to the PRs in the run

Nothing to block merging for most PRs in the run

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/57970 - Squid backport
https://github.com/ceph/ceph/pull/58323 - Reef backport
https://github.com/ceph/ceph/pull/58637 - Quincy backport

**Pull request merged**